src/base/rss/rssdownloadrule.cpp

```diff
@@ -231,6 +261,28 @@ bool DownloadRule::matches(const QString &articleTitle) const
         return false;
     }
 
+    if (m_smartFilter) {
+        QRegExp f(episodeRegex);
+        //See if we can extract an seaon/episode number or date from the title
+        if (f.indexIn(articleTitle) > -1) {
+                //now see if this episode has been downloaded before
+                QString episodeStr = computeEpisodeName(f.cap(1));
+
+                if (!episodeStr.isEmpty()) {
```

Comment:
not a merge blocker here, just too many indentation here, is it possible to refactor it?
something like `if (episodeStr.isEmpty()) return false;` style...